### PR TITLE
Allow aerodyne droppers to hover

### LINKED ISSUE #3386: Aerodyne DropShips Cannot Hover On Low-Altitude Map
### Environment
MegaMek 0.48.0 Windows

### Description
In MegaMek, aerodyne DropShips cannot hover on the Low-Altitude Map. Per _Total Warfare_ p. 84, "If the aerospace fighter, aerodyne small craft or **DropShip** still has unspent thrust points for this turn they may expend thrust to hover, requiring 2 Thrust Points to do so."

### PR DESCRIPTION
**Pull request by pavelbraginskiy**
Fixes #3386

### CODE REVIEW

File: megamek/src/megamek/common/Dropship.java [hunk 1]
```diff
@@ -1869,4 +1869,13 @@ public void newRound(int roundNumber) {
     public void postProcessFacingChange() {
         mpUsed += 2;
     }
+
+    /**
+     * Depsite being able to hover, aerodyne dropships are
+     * explicitely forbidden from vertical landings.
+     */
+    @Override
+    public boolean canLandVertically() {
```
Comment (NickAragua) [inline megamek/src/megamek/common/Dropship.java:1878]: This isn't quite correct: per TW page 88, aerodyne dropships and small may land vertically in vacuum.

**Review by NickAragua — CHANGES_REQUESTED** [review]
Looks good, but needs a small logic adjustment.

**Comment by pavelbraginskiy** [comment]
@NickAragua Made the fix for atmosphere, also fixed the same bug for small craft.

**Review by NickAragua — APPROVED** [review]
Great, thanks for the contribution!

**Comment by pavelbraginskiy** [comment]
@NickAragua I just realized that isVSTOL also affects takeoff/landing distance. 

It might make sense to revert the change and instead add a function like `canHover` that defaults to `isVstol() || isSpheroid()` and override that in the SmallCraft class, or I could override `getTakeOffLength` and `getLandingLength` there.

**Pull request merged**